catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.formulaeditor.UserList
+import java.util.ArrayList
+
+class DeleteItemOfUserListAction : TemporalAction() {
+    private var sprite: Sprite? = null
+    private var formulaIndexToDelete: Formula? = null
+    private var userList: UserList? = null
+
+    override fun update(percent: Float) {
+        if (userList == null) {
+            return
+        }
+        if (userList!!.value.size == 0) {
+            return
+        }
+
+        var indexToDelete: Int = try {
+            if (formulaIndexToDelete == null) {
+                1
+            } else {
+                formulaIndexToDelete!!.interpretInteger(sprite)
+            }
+        } catch (interpretationException: InterpretationException) {
+            Log.e(DeleteItemOfUserListAction::class.java.simpleName,
+                "Interpreting sprite as integer failed : " + sprite.toString(), interpretationException)
+            1
+        }
+
+        indexToDelete--
+
+        if (indexToDelete >= userList!!.value.size || indexToDelete < 0) {
+            return
+        }
+
+        (userList!!.value as ArrayList<Object?>).removeAt(indexToDelete)
+    }
+
+    fun setUserList(userVariable: UserList?) {
+        this.userList = userVariable
+    }
+
+    fun setFormulaIndexToDelete(formulaIndexToDelete: Formula?) {
+        this.formulaIndexToDelete = formulaIndexToDelete
+    }
+
+    fun setSprite(sprite: Sprite?) {
+        this.sprite = sprite
+    }
```

Comment:
```suggestion
```

You can delete all of these. Kotlin will generate them automatically based on member visibility. Also, setters should not accept null.